test/python/transpiler/test_pass_scheduler.py

```diff
@@ -550,6 +550,176 @@ def test_fresh_initial_state(self):
             ],
         )
 
+    def test_nested_conditional_in_loop(self):
+        """Run a loop with a nested conditional."""
+        nested_conditional = [
+            ConditionalController(
+                [PassA_TP_NR_NP()], condition=lambda property_set: property_set["property"]
+            )
+        ]
+        self.passmanager.append(
+            [PassK_check_fixed_point_property(), PassE_AP_NR_NP(True)]
+            + nested_conditional
+            + [PassF_reduce_dag_property()],
+            do_while=lambda property_set: not property_set["property_fixed_point"],
+        )
```

Comment:
It's hard to see what is supposed to happen here. But, it looks like `PassA_TP_NR_NP` runs during every iteration of the while loop. Is this right? If so, it doesn't seem to check whether the conditional works.

Reply:
This has been fixed in a commit made below: 46013e0